internal/pkg/resources/cleanup.go

```diff
@@ -0,0 +1,134 @@
+package resources
+
+import (
+	"context"
+	"errors"
+
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/common"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+)
+
+var ErrDeletionNotFinished = errors.New("deletion is not yet finished")
+
+type ConcurrentCleanup struct {
+	clnt     client.Client
+	policy   client.PropagationPolicy
+	manifest *v1beta2.Manifest
+}
+
+func NewConcurrentCleanup(clnt client.Client, manifest *v1beta2.Manifest) *ConcurrentCleanup {
+	return &ConcurrentCleanup{
+		clnt:     clnt,
+		policy:   client.PropagationPolicy(apimetav1.DeletePropagationBackground),
+		manifest: manifest,
+	}
+}
+
+func (c *ConcurrentCleanup) DeleteDiffResources(ctx context.Context, resources []*resource.Info,
+) error {
+	status := c.manifest.GetStatus()
+	operatorRelatedResources, operatorManagedResources, err := splitResources(resources)
+	if err != nil {
+		return err
+	}
+
+	if err := c.CleanupResources(ctx, operatorManagedResources, status); err != nil {
+		return err
+	}
+
+	if err := c.CleanupResources(ctx, operatorRelatedResources, status); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *ConcurrentCleanup) CleanupResources(
+	ctx context.Context,
+	resources []*resource.Info,
+	status shared.Status,
+) error {
+	if err := c.Run(ctx, resources); errors.Is(err, ErrDeletionNotFinished) {
+		c.manifest.SetStatus(status.WithState(shared.StateWarning).WithErr(err))
+		return err
+	} else if err != nil {
+		c.manifest.SetStatus(status.WithState(shared.StateError).WithErr(err))
+		return err
+	}
+	return nil
+}
+
+func splitResources(resources []*resource.Info) ([]*resource.Info, []*resource.Info, error) {
+	operatorRelatedResources := make([]*resource.Info, 0)
+	operatorManagedResources := make([]*resource.Info, 0)
+
+	for _, resource := range resources {
+		obj, ok := resource.Object.(client.Object)
+		if !ok {
+			return nil, nil, common.ErrTypeAssert
+		}
+		if isOperatorRelatedResources(obj.GetObjectKind().GroupVersionKind().Kind) {
+			operatorRelatedResources = append(operatorRelatedResources, resource)
+			continue
+		}
+		operatorManagedResources = append(operatorManagedResources, resource)
+	}
+
+	return operatorRelatedResources, operatorManagedResources, nil
+}
+
+func isOperatorRelatedResources(kind string) bool {
+	return kind == "CustomResourceDefinition" ||
+		kind == "ServiceAccount" ||
```

Comment:
do you really think the value in the list need to be const, what value it brings? Actually what I expected orignailly is that we can have some const variable can be directly reference from `k8s.io/api` package, to prevent this resource name get changed in future (although in rare cases it will happen), but unforturnately, can't find any. 

But it's good you point out this, I create a unit test to make sure the kind name not get changed, please check.